core/src/main/java/org/elasticsearch/common/lucene/uid/PerThreadIDAndVersionLookup.java

```diff
@@ -51,49 +51,52 @@
     private final TermsEnum termsEnum;
     /** _version data */
     private final NumericDocValues versions;
+
     /** Reused for iteration (when the term exists) */
     private PostingsEnum docsEnum;
 
+    private final Object readerKey;
+
     /**
      * Initialize lookup for the provided segment
      */
     PerThreadIDAndVersionLookup(LeafReader reader) throws IOException {
-        TermsEnum termsEnum = null;
-        NumericDocValues versions = null;
-
         Fields fields = reader.fields();
-        if (fields != null) {
-            Terms terms = fields.terms(UidFieldMapper.NAME);
-            if (terms != null) {
-                termsEnum = terms.iterator();
-                assert termsEnum != null;
-                versions = reader.getNumericDocValues(VersionFieldMapper.NAME);
-                assert versions != null;
-            }
-        }
-
-        this.versions = versions;
-        this.termsEnum = termsEnum;
+        Terms terms = fields.terms(UidFieldMapper.NAME);
```

Comment:
I double checked with Adrien and this is OK (and is something he wanted to do for a long time).